### DISCUSSION TITLE
`PlainTime.prototype.toString()` includes seconds by default. Possible to set `smallestUnit: "minutes"` as a global default?

### DISCUSSION THREAD

**Discussion by revosw** [js-temporal/temporal-polyfill#260] [Q&A] [answered]
In my application, I exclusively work with hours and minutes, never with seconds. My own web APIs take strings in `HH:MM` form, which means I have to serialize with `.toString({ smallestUnit: "minutes" })` all the time.

Is it possible to set a global default? That would make it possible to write code like this:
```js
const appointment = {
  start: Temporal.PlainTime.from("09:00"),
  duration: "PT15M"
}

// 😀 short and sweet
await createAppointment(appointment)
// 🙁 sent to the api: { start: "09:00:00", duration: "PT15M" }
```
Instead of like this:
```js
// 🙁 long and wordy
await createAppointment({ start: appointment.start.toString({ smallestUnit: "minutes" }), duration: appointment.duration })
// 😀 sent to the api: { start: "09:00", duration: "PT15M" }
```

**Comment by justingrant** [comment, marked as answer]
Thanks for the question!  No, there is no global default. ECMAScript generally avoids global state, because one user's changes can affect other code. 

But here's a few possible ways you could customize things to get what you want:
* You could use a custom `JSON.stringify` [replacer function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#the_replacer_parameter). 
* You could patch `Temporal.prototype.toJSON`, but there's danger there because if your code calls into other libraries that use Temporal, then your patching could break them. 
* You could build an API wrapper 

Good luck with your project!

**Reply by revosw** [reply]
Thanks for the reply! I'll have a look at making an API wrapper of some sort
